lib/code_ownership/private/team_plugins/github.rb

```diff
@@ -22,13 +22,22 @@ def github
         sig { override.params(teams: T::Array[CodeTeams::Team]).returns(T::Array[String]) }
         def self.validation_errors(teams)
           all_github_teams = teams.flat_map { |team| self.for(team).github.team }
+          missing_github_teams = teams.select { |team| self.for(team).github.team.nil? }
```

Comment:
This change makes sense, but one concern is that this is technically a breaking change since it was previously allowed to be `nil` and now it is no longer allowed to be.

One alternative to simplify this is to just ignore teams that do not have this set when specifying teams used more than once. That way it wouldn't be considered an error and wouldn't show up in the confusing way you described in the PR description.

Let me know what ya think @timlkelly

Reply:
Great call out! Thanks for bringing that up. Is there interest in a separate team plugin to require the Github key? Then it could potentially be opt-in and a non-breaking change.

Reply:
That could be a separate plugin. Another possibility is if the code ownership configuration YML file could support an optional configuration flag, e.g. `require_github_teams` that errors if any are nil. That way it can be opt-in without needing to fragment the plugin.

Reply:
@alexevanczuk good idea. I've updated it to include a new configuration key. I'm new to sorbet, so let me know if I missed anything there.